src/prop/cadical.cpp

```diff
@@ -948,16 +951,30 @@ class CadicalPropagator : public CaDiCaL::ExternalPropagator
 
 CadicalSolver::CadicalSolver(Env& env,
                              StatisticsRegistry& registry,
-                             const std::string& name)
+                             const std::string& name,
+                             bool logProofs)
     : EnvObj(env),
       d_solver(new CaDiCaL::Solver()),
       d_context(nullptr),
       // Note: CaDiCaL variables start with index 1 rather than 0 since negated
       //       literals are represented as the negation of the index.
       d_nextVarIdx(1),
+      d_logProofs(logProofs),
       d_inSatMode(false),
       d_statistics(registry, name)
 {
+  if (logProofs)
+  {
+    std::stringstream ssp;
+    ssp << options().driver.filename << ".drat_proof.txt";
+    d_pfFile = ssp.str();
```

Comment:
Simplify and avoid unnecessary creation of stringstream and subsequent string copy  with
```suggestion
    d_pfFile = options().driver.filename + ".drat_proof.txt";
```